src/DragHandle.tsx

```diff
@@ -19,6 +22,106 @@ const cellDragHandle = css`
 
 const cellDragHandleClassname = `rdg-cell-drag-handle ${cellDragHandle}`;
 
-export default function DragHandle(props: React.HTMLAttributes<HTMLDivElement>) {
-  return <div className={cellDragHandleClassname} {...props} />;
+interface Props<R, SR> extends Pick<DataGridProps<R, SR>, 'rows' | 'onRowsChange'> {
+  columns: readonly CalculatedColumn<R, SR>[];
+  selectedPosition: SelectCellState | EditCellState<R>;
+  onFill: (event: FillEvent<R>) => R[];
+  setDragging: (isDragging: boolean) => void;
+  setDraggedOverRowIdx: React.Dispatch<React.SetStateAction<number | undefined>>;
+}
+
+export default function DragHandle<R, SR>({
+  rows,
+  columns,
+  selectedPosition,
+  onRowsChange,
+  onFill,
+  setDragging,
+  setDraggedOverRowIdx
+}: Props<R, SR>) {
+  function handleMouseDown(event: React.MouseEvent<HTMLDivElement, MouseEvent>) {
+    if (event.buttons !== 1) return;
+    setDragging(true);
+    window.addEventListener('mouseover', onMouseOver);
+    window.addEventListener('mouseup', onMouseUp);
+
+    function onMouseOver(event: MouseEvent) {
+      // Trigger onMouseup in edge cases where we release the mouse button but `mouseup` isn't triggered,
+      // for example when releasing the mouse button outside the iframe the grid is rendered in.
+      // https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons
+      if (event.buttons !== 1) onMouseUp();
+    }
+
+    function onMouseUp() {
+      window.removeEventListener('mouseover', onMouseOver);
+      window.removeEventListener('mouseup', onMouseUp);
+      setDragging(false);
+      handleDragEnd();
+    }
+  }
+
+  function handleDragEnd() {
+    if (!onRowsChange) return;
```

Comment:
Should disable the feature if `onRowsChange` isn't defined

Reply:
I think we should allow it if `onFill != null`. It is like editor, we do not block editor if `onRowsChange` is not specified.

Reply:
`onFill` cannot be called when `onRowsChange` is nullish though right now, should remove the return guards.